pkg/schedule/scatter/region_scatterer.go

```diff
@@ -287,6 +290,12 @@ func (r *RegionScatterer) Scatter(region *core.RegionInfo, group string, skipSto
 		return nil, errors.Errorf("region %d has no leader", region.GetID())
 	}
 
+	// Check if region is in an affinity group that doesn't allow regular scheduling
+	if !r.affinityFilter.Select(region).IsOK() {
```

Comment:
If it returns an error, the client will retry the scatter, but it always fails. So we should return nil for the affinity region

Reply:
ok, I fixed it in https://github.com/tikv/pd/pull/10091#issuecomment-3673141192

Reply:
done